File: backend/src/models/bill.rs
use sqlx::FromRow;
use serde::{Deserialize, Serialize};
use chrono::NaiveDate;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Bill {
    pub id: i32,
    pub form_no: Option<String>,
    pub serial_no: Option<String>,
    pub invoice_no: Option<String>,
    pub issued_date: Option<NaiveDate>,
    pub seller_name: Option<String>,
    pub seller_tax_code: Option<String>,
    pub item_name: Option<String>,
    pub unit: Option<String>,
    pub quantity: Option<rust_decimal::Decimal>,
    pub unit_price: Option<rust_decimal::Decimal>,
    pub total_amount: Option<rust_decimal::Decimal>,
    pub vat_rate: Option<rust_decimal::Decimal>,
    pub vat_amount: Option<rust_decimal::Decimal>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateBill {
    pub form_no: Option<String>,
    pub serial_no: Option<String>,
    pub invoice_no: Option<String>,
    pub issued_date: Option<NaiveDate>,
    pub seller_name: Option<String>,
    pub seller_tax_code: Option<String>,
    pub item_name: Option<String>,
    pub unit: Option<String>,
    pub quantity: Option<rust_decimal::Decimal>,
    pub unit_price: Option<rust_decimal::Decimal>,
    pub total_amount: Option<rust_decimal::Decimal>,
    pub vat_rate: Option<rust_decimal::Decimal>,
    pub vat_amount: Option<rust_decimal::Decimal>,
}

File: backend_python/models/bill.py
from datetime import date
from decimal import Decimal

from sqlalchemy import Date, Numeric, Text
from sqlalchemy.orm import Mapped, mapped_column

from .base import Base


class Bill(Base):
    __tablename__ = "bills"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    form_no: Mapped[str | None] = mapped_column(Text, nullable=True)
    serial_no: Mapped[str | None] = mapped_column(Text, nullable=True)
    invoice_no: Mapped[str | None] = mapped_column(Text, nullable=True)
    issued_date: Mapped[date | None] = mapped_column(Date, nullable=True)
    seller_name: Mapped[str | None] = mapped_column(Text, nullable=True)
    seller_tax_code: Mapped[str | None] = mapped_column(Text, nullable=True)
    item_name: Mapped[str | None] = mapped_column(Text, nullable=True)
    unit: Mapped[str | None] = mapped_column(Text, nullable=True)
    quantity: Mapped[Decimal | None] = mapped_column(Numeric(18, 2), nullable=True)
    unit_price: Mapped[Decimal | None] = mapped_column(Numeric(18, 2), nullable=True)
    total_amount: Mapped[Decimal | None] = mapped_column(Numeric(18, 2), nullable=True)
    vat_rate: Mapped[Decimal | None] = mapped_column(Numeric(5, 2), nullable=True)
    vat_amount: Mapped[Decimal | None] = mapped_column(Numeric(18, 2), nullable=True)
